VorpalEngine/sources/video/Renderer.cpp

```diff
@@ -164,6 +429,597 @@ bool Renderer::CreateInstance()
+            return false;
+        }
+    }
+    return true;
+}
+
+bool Renderer::CreateRenderPass()
+{
+    if(m_renderPass != VK_NULL_HANDLE)
+    {
+        vkDestroyRenderPass(m_vkLogicalDevice, m_renderPass, nullptr);
+        m_renderPass = VK_NULL_HANDLE;
+    }
+    VkAttachmentDescription colorAttachment = {};
+    colorAttachment.format = m_swapChainImageFormat;
+    colorAttachment.samples = VK_SAMPLE_COUNT_1_BIT;
+    colorAttachment.loadOp = VK_ATTACHMENT_LOAD_OP_CLEAR;
+    colorAttachment.storeOp = VK_ATTACHMENT_STORE_OP_STORE;
+    colorAttachment.stencilLoadOp = VK_ATTACHMENT_LOAD_OP_DONT_CARE;
+    colorAttachment.stencilStoreOp = VK_ATTACHMENT_STORE_OP_DONT_CARE;
+    colorAttachment.initialLayout = VK_IMAGE_LAYOUT_UNDEFINED;
+    colorAttachment.finalLayout = VK_IMAGE_LAYOUT_PRESENT_SRC_KHR;
+
+    VkAttachmentReference colorAttachmentRef = {};
+    colorAttachmentRef.attachment = 0;
+    colorAttachmentRef.layout = VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL;
+
+    VkSubpassDescription subpass = {};
+    subpass.pipelineBindPoint = VK_PIPELINE_BIND_POINT_GRAPHICS;
+    subpass.colorAttachmentCount = 1;
+    subpass.pColorAttachments = &colorAttachmentRef;
+
+    VkRenderPassCreateInfo renderPassInfo = {};
+    renderPassInfo.sType = VK_STRUCTURE_TYPE_RENDER_PASS_CREATE_INFO;
+    renderPassInfo.attachmentCount = 1;
+    renderPassInfo.pAttachments = &colorAttachment;
+    renderPassInfo.subpassCount = 1;
+    renderPassInfo.pSubpasses = &subpass;
+
+    if (vkCreateRenderPass(m_vkLogicalDevice, &renderPassInfo, nullptr, &m_renderPass) != VK_SUCCESS) {
+       LOG_ERROR("Failed to create render pass!");
+       return false;
+    }
+
+    return true;
+}
+
+bool Renderer::CreateGraphicsPipeline()
+{
+    if(m_pipelineLayout != VK_NULL_HANDLE)
+    {
+        vkDestroyPipelineLayout(m_vkLogicalDevice, m_pipelineLayout, nullptr);
+        m_pipelineLayout = VK_NULL_HANDLE;
+    }
+    if(m_graphicsPipeline != VK_NULL_HANDLE)
+    {
+        vkDestroyPipeline(m_vkLogicalDevice, m_graphicsPipeline,nullptr);
+        m_graphicsPipeline = VK_NULL_HANDLE;
+    }
+
+    //TODO: real loading code.
+    //    vp::utility::asset::SimpleStorage& storage = vp::utility::asset::SimpleStorage::Instance();
+    //    vp::utility::asset::Handler simpleVertShaderHandler = storage.Get("data/shaders/vert.spv");
+    //    vp::utility::asset::Handler simpleFragShaderHandler = storage.Get("data/shaders/frag.spv");
+    //    if(!simpleVertShaderHandler.IsValid()
+    //            || !simpleFragShaderHandler.IsValid()) {
+    //        LOG_ERROR("Can't find shaders!");
+    //        return false;
+    //    }
+
+    auto vertShaderCode = readFile("data/shaders/vert.spv");
+    auto fragShaderCode = readFile("data/shaders/frag.spv");
+    VkShaderModule vertShaderModule = {};
+    VkShaderModule fragShaderModule = {};
+    bool shadersCreated = !CreateShaderModule(vertShaderCode, vertShaderModule)
```

Comment:
variable is named `shadersCreated` but it holds invert of that statement. Rename?